src/Serializers/VariantSerializer.js

```diff
@@ -27,7 +27,9 @@ class VariantSerializer {
         const decoded = RLP.decode(buffer.slice(1), true)
         const arities = [...decoded.data]
         const tag = decoded.remainder[0]
-        const [els, rest] = this.globalSerializer.deserializeStream(decoded.remainder.slice(1))
+        const valueType = typeof typeInfo === 'undefined' ? undefined
+            : { ...typeInfo.variantType, valueTypes: Object.values(typeInfo.variants[tag])[0] }
```

Comment:
this is too long and unreadable for ternary if, please use it only for single liners without complex expressions

else is also not needed most of the times in general

Reply:
also that check already exists below, so it can be combined:
```javascript
    deserializeStream(data, typeInfo) {
        const buffer = new Uint8Array(data)
        const decoded = RLP.decode(buffer.slice(1), true)
        const arities = [...decoded.data]
        const tag = decoded.remainder[0]

        let variants = []
        let valueType = undefined
        if (typeof typeInfo != 'undefined') {
            variants = typeInfo.variants
            valueType = {
                ...typeInfo.variantType,
                valueTypes: Object.values(variants[tag])[0]
            }
        }

        const [els, rest] = this.globalSerializer.deserializeStream(decoded.remainder.slice(1), valueType)

        return [
            new FateVariant(arities, tag, els.items, els.valueTypes, variants),
            rest
        ]
    }
```